botCommands/leaderboard.js

```diff
@@ -10,30 +10,31 @@ const leaderboard = {
         .split(' ')
         .find((word) => word.includes('start='));
       let start = sEquals ? sEquals.replace('start=', '') : 1;
-      start = Math.max(start, 1);
+      start = Number.isNaN(Number(start)) ? 1 : Math.max(start, 1);
 
       const nEquals = content.split(' ').find((word) => word.includes('n='));
       let length = nEquals ? nEquals.replace('n=', '') : 5;
+      length = Number.isNaN(Number(length)) ? 5 : length;
       length = Math.min(length, 25);
       length = Math.max(length, 1);
 
-      const users = await axios.get(
-        'https://www.theodinproject.com/api/points',
+      const { data: users } = await axios.get(
+        `https://www.theodinproject.com/api/points?limit=${length}&offset=${start - 1}`,
       );
-      const data = users.data.filter((user) => guild.members.cache.get(user.discord_id));
```

Comment:
I get why you've removed this line where we validate if the user exists in the guild. But without this filtering, we'll end up displaying "UNDEFINED" in our leaderboard. This is what I mean:

![image](https://user-images.githubusercontent.com/85733202/212676669-13a3a98e-a70d-41f4-8ef3-4845168ffaa4.png)


Even if we keep this filter, it won't have a huge effect because we're always fetching a subset of the leaderboard. One way to tackle this would be to still display the points and rank on the leaderboard with "User left the server" instead of username. Or maybe not do anything at all? I'll also discuss with the team. Do share your thoughts, thanks!

Reply:
Yeah that was very much the purpose of this [comment](https://github.com/TheOdinProject/odin-bot-v2/pull/256#discussion_r1071171702) earlier. 
Perhaps we could trigger an API call that removes the user details from the API once they leave or are banned from the guild?